### PR TITLE
Add package sample

### PR DESCRIPTION
**Pull request by johnpryan**
This is a sample of a Dart package to be featured in an upcoming talk.

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]
Please add this to CI

**Comment by johnpryan** [comment]
I'd prefer to leave the CI in [constraint_solver.yaml](https://github.com/dart-lang/samples/pull/174/files#diff-50a12f7583e81c9a9b501c7f9797bb33acb85a5ca795b76dc8a773847b571c30), since that's part of the content for the talk.

**Comment by domesticmouse** [comment]
@johnpryan feel free to land when you are happy with the code.

File: package_constraint_solver/pubspec.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: constraint_solver
+description: A general-purpose backtracking constraint satisfaction algorithm.
+version: 1.0.0
+repository: https://github.com/johnpryan/constraint_solver
+
+environment:
+  sdk: '>=2.19.2 <4.0.0'
```
Comment (jonasfj) [inline package_constraint_solver/pubspec.yaml:7]: In the future we probably want to make this:
```
  sdk: '^3.0.0'
```

**Review by jonasfj — APPROVED** [review]

**Pull request merged**